state-chain/pallets/cf-ingress-egress/src/tests/boost.rs

```diff
@@ -795,31 +795,20 @@ fn test_add_boost_funds() {
 }
 
 #[track_caller]
-fn turn_safe_mode_on() {
-	assert!(
-		<MockRuntimeSafeMode as sp_core::Get<PalletSafeMode<()>>>::get() ==
-			PalletSafeMode::CODE_GREEN,
-		"Safe mode was already on"
-	);
-	<MockRuntimeSafeMode as SetSafeMode<MockRuntimeSafeMode>>::set_code_red();
-	assert!(
-		<MockRuntimeSafeMode as sp_core::Get<PalletSafeMode<()>>>::get() ==
-			PalletSafeMode::CODE_RED
-	);
-}
+fn toggle_boosting_with_safe_mode(enable: bool) {
+	fn get_safe_mode() -> PalletSafeMode<()> {
+		<MockRuntimeSafeMode as sp_core::Get<PalletSafeMode<()>>>::get()
+	}
 
-#[track_caller]
-fn turn_safe_mode_off() {
-	assert!(
-		<MockRuntimeSafeMode as sp_core::Get<PalletSafeMode<()>>>::get() ==
-			PalletSafeMode::CODE_RED,
-		"Safe mode was already off"
-	);
-	<MockRuntimeSafeMode as SetSafeMode<MockRuntimeSafeMode>>::set_code_green();
-	assert!(
-		<MockRuntimeSafeMode as sp_core::Get<PalletSafeMode<()>>>::get() ==
-			PalletSafeMode::CODE_GREEN
-	);
+	let boost_mode = if enable { PalletSafeMode::CODE_GREEN } else { PalletSafeMode::CODE_RED };
+
+	let new_mode =
+		PalletSafeMode { deposits_enabled: get_safe_mode().deposits_enabled, ..boost_mode };
```

Comment:
This is ok, but i think it would be better to just set the individual bool's in the test, like what you did in the `safe_mode_prevents_deposit`. That way we can make sure we are using the correct flag as well. To keep it clean we can still use the old `turn_safe_mode_off` function.

Reply:
I think right now it is not clear which approach is better. When we have more bools in `PalletSafeMode`, we will have a better idea, and maybe we could revisit then.